lib/SIL/Linker.cpp

```diff
@@ -218,7 +219,7 @@ bool SILLinkerVisitor::visitProtocolConformance(
       // to deserialize everything, add the function to the list of functions
       // to deserialize.
       performFuncDeserialization = true;
-      addFunctionToWorklist(E.getMethodWitness().Witness);
+      maybeAddFunctionToWorklist(E.getMethodWitness().Witness);
```

Comment:
You are also expanding the conditions here. Same concern: what about specialized witness tables? That being said, maybe this is not interesting.

Reply:
The only way we can end here in the mandatory pipeline (where isLinkAll is off) is for imported conformances, where all witnesses are shared. So again there’s no real change here. We are just avoiding adding functions to the work list if they already have a body.